src/compiler/checker.ts

```diff
@@ -7084,6 +7084,14 @@ namespace ts {
                 return addOptionality(declaredType, isOptional);
             }
 
+            const jsDocTypeDefTag = getJSDocTypedefTag(declaration);
+            // The typedef is attached to a variable declaration, instead of specifying the name in the typedef itself
+            // This means that the variable declared references the typedef and should not be used in regular control
+            // flow of the program, only in jsdoc type references.
+            if (jsDocTypeDefTag && !jsDocTypeDefTag.name && isVariableDeclaration(declaration)) {

```

Comment:
probably need to check that there is no initialiser too

Reply:
oh, and `isInJSFile`, no exports, not a binding pattern, not a `declare` (which is illegal in js, but might show up anyway).